aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -160,12 +160,13 @@ impl IndyCredxAnonCreds {
 
         let wql_query = if let Some(restrictions) = restrictions {
             match restrictions {
-                Value::Array(mut arr) => {
-                    arr.extend(attrs);
-                    json!({ "$and": arr })
+                Value::Array(restrictions) => {
+                    let restrictions_wql = json!({ "$or": restrictions });
+                    attrs.push(restrictions_wql);
+                    json!({ "$and": attrs })
                 }
-                Value::Object(obj) => {
-                    attrs.push(Value::Object(obj));
+                Value::Object(restriction) => {
+                    attrs.push(Value::Object(restriction));
                     json!({ "$and": attrs })
                 }
                 _ => json!(attrs),
```

Comment:
I suppose this case is us saying "ignore the restrictions if they aren't a valid restriction json type (obj or array)".

I wonder more generally if we should just throw an error here; "unsupported restriction". 

If not, I'm not sure json!(attrs) is correct for ignoring restrictions.

I think it should be json!({$and: attrs})

To indicate that ALL attr in the attrs array must be satisfied

Reply:
Also a couple lines below (175) has the same issue

Reply:
should be addressed, please check

Reply:
Sorry for delay, your fox matches my understanding, thanks